packages/harmony/src/components/input/TextInput/types.ts

```diff
@@ -0,0 +1,86 @@
+import type { ComponentPropsWithoutRef } from 'react'
+
+import type { IconComponent } from '../../typography'
+
+export enum TextInputSize {
+  SMALL = 'small',
+  DEFAULT = 'default'
+}
+
+export type TextInputProps = Omit<
+  ComponentPropsWithoutRef<'input'>,
+  // Omitting required purely for storybook docs
+  'size' | 'required'
+> & {
+  /**
+   * Input sizes. NOTE: small inputs will not show the label
+   * @default default
+   */
+  size?: TextInputSize
+  /**
+   * Toggles warning state (turns border warning color). NOTE: this is not used any where at the moment
+   */
+  warning?: boolean
+  /**
+   * Toggles error state
+   */
+  error?: boolean
+  /**
+   * Hides the label. If the label is hidden the placeholder will show instead.
+   * @default false
+   */
+  hideLabel?: boolean
+  /**
+   * ClassName on the div wrapping the whole input container (doesn't include assistive text)
+   */
+  inputRootClassName?: string
+  /**
+   * Helper text that shows up below the input
+   */
+  helperText?: string
+  /**
+   * Floating text on the lefthand side of the input.
+   */
+  startAdornmentText?: string
+  /**
+   * Floating text on the righthand side of the input
+   */
+  endAdornmentText?: string
+  /**
+   * Floating icon on the lefthand side of the input. Note: will float to the left of the label & content
+   */
+  startIcon?: IconComponent
+  /**
+   * Floating text on the righthand side of the input
+   */
+  endIcon?: IconComponent
+  /**
+   * Required or not. Will add an * to the label if required
+   */
+  required?: boolean
+  /**
+   * 0-1 number representating a percentage threshold to show the max character text. Default is 0.7 (70%)
+   * @default 0.7
+   */
+  showMaxLengthThreshold?: number
+  /**
+   * 0-1 number representating a percentage threshold to turn the character limit text orange. Default is 0.9 (90%)
+   * @default 0.9
+   */
+  maxLengthWarningThreshold?: number
+} & ( // Make either label or aria-label required prop
+    | {
+        /**
+         * Label Text (Required if aria-label is not provided)
+         */
+        label: string
+        ['aria-label']?: string
+      }
+    | {
+        /**
+         * Label Text (Required if aria-label is not provided)
+         */
+        ['aria-label']: string
+        label?: string
+      }
+  )
```

Comment:
Would love some feedback on this, put this in here as kind of a "proposal". 
This effectively makes providing either `label={}` or `aria-label={}` required. A bit heavy handed but makes sure we always solve for accessibility. 
The only part that I really don't like is that the TS error isn't very helpful which might make this not worth 😅 
I'm open to whatever others think
![image](https://github.com/AudiusProject/audius-protocol/assets/6711655/2c8a78f8-b10e-44c5-a14d-b7579d0f264c)

Reply:
I like your thinking, my hope is that we assign this to the size of the input, like if it's a small input, aria-label is required, and for large label is required and aria-label optional or maybe not allowed?

Reply:
Ooh this is a tough one. I'm tempted to say we should just require label and put a comment on the type saying it's required for accessibility

Reply:
ahh so if you provide label, then it becomes the aria-label for the smaller input? i like that a lot

Reply:
Yeah I agree, I'm gonna do `label` required regardless of whether it's shown visually or not. 
I thought it was being slick with that union setup until I saw that error message 😂

Reply:
This might be helpful here! https://gist.github.com/palashmon/db68706d4f26d2dbf187e76409905399